hub3/models/dataset_test.go

```diff
@@ -14,196 +14,202 @@
-			err := ORM().All(&ds)
-			Expect(err).To(BeNil())
-			Expect(len(ds)).To(Equal(1))
-		})
-
-		It("should be able to find it in the database", func() {
-			var ds DataSet
-			err := ORM().One("Spec", spec, &ds)
-			Expect(err).To(BeNil())
-			Expect(ds.Created.Unix()).To(Equal(dataset.Created.Unix()))
-			Expect(ds.Modified.UnixNano()).ToNot(Equal(dataset.Modified.UnixNano()))
-			Expect(ds.Access.LOD).To(BeTrue())
-		})
-
-	})
-
-	Context("When calling CreateDataSet", func() {
-
-		It("should create a dataset when no dataset is present.", func() {
-			ds, created, err := CreateDataSet("test3")
-			Expect(err).ToNot(HaveOccurred())
-			Expect(created).To(BeTrue())
-			Expect(ds.Spec).To(Equal("test3"))
-		})
-
-	})
-
-	Context("When calling GetOrCreateDataSet", func() {
-
-		It("should create the datasets when no dataset is available", func() {
-			ds, created, err := GetOrCreateDataSet("test2")
-			Expect(err).ToNot(HaveOccurred())
-			Expect(created).To(BeTrue())
-			Expect(ds.Spec).To(Equal("test2"))
-		})
-
-		It("should not store the dataset again on Get", func() {
-			datasetCount, err := ListDataSets()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(len(datasetCount) > 0).To(BeTrue())
-			ds, created, err := GetOrCreateDataSet("test2")
-			Expect(err).ToNot(HaveOccurred())
-			Expect(created).To(BeFalse())
-			Expect(ds.Spec).To(Equal("test2"))
-			newCount, err := ListDataSets()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(len(datasetCount)).To(Equal(len(newCount)))
-		})
-	})
-
-	Context("When calling ListDatasets", func() {
-
-		It("should return an array of all stored datasets", func() {
-			datasets, err := ListDataSets()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(datasets).ToNot(BeEmpty())
-		})
-
-	})
-
-	Context("When calling IncrementRevision", func() {
-
-		It("should update the revision of the dataset by one", func() {
-			ds, _, _ := GetOrCreateDataSet("test3")
-			Expect(ds.Revision).To(Equal(1))
-			_, err := ds.IncrementRevision()
-			Expect(err).ToNot(HaveOccurred())
-			ds, _, _ = GetOrCreateDataSet("test3")
-			Expect(ds.Revision).To(Equal(2))
-		})
-
-		It("should have stored the dataset with the new revision", func() {
-			ds, _, _ := GetOrCreateDataSet("test3")
-			Expect(ds.Revision).To(Equal(2))
-		})
-	})
-
-	// todo add code for removing datasets.
-	Context("When calling delete", func() {
-
-		It("should delete the dataset", func() {
-			dataSets, err := ListDataSets()
-			Expect(err).To(BeNil())
-			dsNr := len(dataSets)
-			dsName := "test4"
-			ds, _, _ := GetOrCreateDataSet(dsName)
-			Expect(ds).ToNot(BeNil())
-			ds, err = GetDataSet(dsName)
-			Expect(err).To(BeNil())
-			ctx := context.Background()
-			wp := workerpool.New(1)
-			err = ds.Delete(ctx, wp)
-			Expect(err).To(BeNil())
-			ds, err = GetDataSet(dsName)
-			Expect(err).ToNot(BeNil())
-			dataSets, err = ListDataSets()
-			Expect(err).To(BeNil())
-			Expect(dsNr).To(Equal(len(dataSets)))
-		})
-
-	})
-
-})
+// import (
```

Comment:
When will the test return?

Reply:
This will  be migrated to the dataset service in another pull